### PR TITLE
feat: support serdeTuple decorator

### PR DESCRIPTION
**Pull request by yjhmelody**
- Clean up so many code for better reading.
- Support serdeTuple decorator.
- Though we have also impl serializeTuple/deserializeTuple decorator, but now we disable them.

### CODE REVIEW

File: packages/transform/src/tests/visitors/common.ts [hunk 1]
```diff
@@ -1,31 +1,22 @@
...
-    serdeKind: ClassSerdeKind,
 ): void {
     const parser = new Parser();
     parser.parseFile(code, "index.ts", true);
     const source = visitor.visit(parser.sources[0]) as Source;
 
     let stmt = source.statements[0];
     assert.equal(stmt.kind, NodeKind.ClassDeclaration);
-    assert.ok(utils.hasDecorator(stmt as ClassDeclaration, serdeKind));
+    // assert.ok(utils.hasDecorator(stmt as ClassDeclaration, serdeKind));
```
Comment (lambda-world) [inline packages/transform/src/tests/visitors/common.ts:19]: Clean up code

File: packages/transform/src/tests/visitors/common.ts [hunk 2]
```diff
@@ -47,4 +38,5 @@ export enum Case {
     }`,
     EmptyBar = "class Bar {}",
     EmptyBarExtendsFoo = "class Bar extends Foo {}",
+    MissingField = "class Bar { b = false; }",
```
Comment (lambda-world) [inline packages/transform/src/tests/visitors/common.ts:41]: Better to rename it `MissingFieldType`

File: packages/transform/src/tests/visitors/common.ts [hunk 3]
```diff
@@ -1,31 +1,21 @@
...
-    NodeKind,
-} from "assemblyscript/dist/assemblyscript.js";
-
+import { ASTBuilder, TransformVisitor } from "visitor-as";
+import { Parser, DiagnosticEmitter, Source, NodeKind } from "assemblyscript/dist/assemblyscript.js";
 import { hasErrorMessage, hasWarningMessage } from "../../utils.js";
-import { ClassSerdeKind } from "../../consts.js";
 
-export function checkVisitor(
+export function commnCheckVisitor(
```
Comment (iamizumi) [inline packages/transform/src/tests/visitors/common.ts:6]:
```suggestion
export function commonCheckVisitor(
```

**Review by lambda-world — APPROVED** [review]
LGTM

**Pull request merged**